diff --git a/lib/zip.h b/lib/zip.h
index 27141b3..7b243bc 100644
--- a/lib/zip.h
+++ b/lib/zip.h
@@ -37,7 +37,7 @@
 
 #ifndef ZIP_EXTERN
 # ifndef ZIP_STATIC
-#  ifdef _WIN32
+#  if defined(_WIN32) && defined(PHP_ZIP_EXPORTS)
 #   define ZIP_EXTERN __declspec(dllimport)
 #  elif defined(__GNUC__) && __GNUC__ >= 4
 #   define ZIP_EXTERN __attribute__ ((visibility ("default")))
diff --git a/lib/zipint.h b/lib/zipint.h
index 40f75c4..323b3f1 100644
--- a/lib/zipint.h
+++ b/lib/zipint.h
@@ -34,8 +34,12 @@
   IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */
 
-#ifdef HAVE_CONFIG_H
-#include "config.h"
+#ifdef PHP_WIN32
+# include "php_zip_config.w32.h"
+#else
+# ifdef HAVE_CONFIG_H
+#  include "config.h"
+# endif
 #endif
 
 /* to have *_MAX definitions for all types when compiling with g++ */
@@ -43,10 +47,10 @@
 
 #include <zlib.h>
 
-#ifdef _WIN32
-#define ZIP_EXTERN __declspec(dllexport)
+#ifdef PHP_WIN32
 /* for dup(), close(), etc. */
 #include <io.h>
+#include "config.w32.h"
 #endif
 
 #ifndef _ZIP_COMPILING_DEPRECATED
diff --git a/lib/zipwin32.h b/lib/zipwin32.h
index 60f8d0c..4fa29cc 100644
--- a/lib/zipwin32.h
+++ b/lib/zipwin32.h
@@ -34,8 +34,6 @@
   IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */
 
-/* 0x0501 => Windows XP; needs to be at least this value because of GetFileSizeEx */
-#define _WIN32_WINNT 0x0501
 #include <windows.h>
 
 /* context for Win32 source */
